InvenTree/label/api.py

```diff
@@ -201,21 +204,97 @@ def print(self, request, items_to_print):
         label_names = []
         label_instances = []
 
-        # Merge one or more PDF files into a single download
-        for item in items_to_print:
+        if not self.get_object().multipage:
```

Comment:
This method is now getting very complicated to read with the different options:

- Plugin printing
- Multi-page printing
- Debug

I'd like to see it broken into separate methods to improve readability and maintainability

Reply:
Also machine printing will extend this - currently implementing it (ref: #4789 )

Reply:
@SchrodingersGat I separated the method to single and multiple labels per page. After some further meditation I still think that the pagination should be done in the plugins (if they are involved).